### ISSUE TITLE
Working on a Project, suddenly mutes some parts in songmode. Windows 10 Pro

### ISSUE THREAD

**Issue by SaschaH87** [hydrogen-music/hydrogen#831]
Hydrogen version *          : 0.9.7 + 1.0.0 Beta 1 + 1.0.0 Beta 2
Operating system + version  : Windows 10 Pro
Audio driver + version      : PortAudio Driver

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Hello,
as described in the topic, i'm working on a song and suddenly when i want to hear the whole thing 
the program seems to mute tiny parts in the songmode.
First i tried the 0.9.7 version then the 1.0.0 Beta 1 version and now the 1.0.0 Beta 2 version.
I have always the same problem with all of the versions.
I hope someone can help me with that, cause otherwise the whole Hydrogen Drum Machine is totally great and i like it alot.

Thanks
Sascha

< describe your issue or feature request here ** >

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

*  H2 version : see 'Info - About', or in case you build Hydrogen from source use 'git rev-parse HEAD'
** if you are reporting a bug, please describe how the bug can be reproduced

**Comment by mauser** [comment]
Hi! 

do you have a way to reproce this problem? What do you mean exactly by "mute tiny parts"?

**Comment by SaschaH87** [comment]
Hi,

when i press the play button in song mode it seems to skip some parts although i can see a pattern as a blue square. it seems to play some patterns just a bit longer than they should.
Sorry but for me thats a bit difficult to explain. I hope this helps a bit.
But i'm trying to give you more information if you have more questions.

I think it is not muting but makes the pattern a bit longer on some parts, but then you can't hear anything cause i don't programmed notes on that point. It makes it somekind of automatically.

Thanks

**Comment by trebmuh** [comment]
Any way to record a screencast demonstrating the thing maybe?

**Comment by SaschaH87** [comment]
Ok i made a screencast video for you where you hopefully can see the problem.

https://streamable.com/8artou

Thanks for your help.

**Comment by elpescado** [comment]
Is it deterministic? I.e. does it occur in the same moment every time?

You might open View Menu and show Automation Path view and see whether there's some drop in velocity there.

Maybe you have a pattern that is longer than other patterns, that might cause other patterns that are concurrent to be longer too.

**Comment by trebmuh** [comment]
Wow, I've never encountered that!
@SaschaH87 after answering @elpescado 's question, can you please provides us with the .h2song from this song please. Thank you.

**Comment by SaschaH87** [comment]
I uploaded the .h2song data for you

https://www.transfernow.net/mQmYiv042020

Thanks

**Comment by trebmuh** [comment]
...but you didn't answer @elpescado 's question first!

**Comment by mauser** [comment]
The problem is your pattern "Crash". Its length is 8, whereas your D-Beat Pattern has a length of 4. Therefore there is a gap at the end of the pattern, so you hear silence. If the crash pattern had more notes, you would hear those notes then. 

But i'm a little bit puzzled why this silence does not apper for the first two time the "crash" pattern is being played..

**Comment by trebmuh** [comment]
Yes, the strange thing is the bug only appears on patterns n°11, 15, and 19. Strangely, not on patterns n°3 and 7.

@SaschaH87 as a workaround, just get the Crash pattern "length" down to 4 and this issue will be workarounded for you.

**Comment by SaschaH87** [comment]
Sorry, i will do now.

I think it occurs on the same moment every time.
It occurs on second 13 and i think at second 18.
I don't think i have a drop in velocity there, but i have different types of pattern some are longer and some shorter, but i thought that is normal in a drum computer?

Thanks for your help

**Comment by trebmuh** [comment]
Yes, the strange thing is the bug only appears on patterns n°11, 15, and 19. Strangely, not on patterns n°3 and 7.

@SaschaH87 as a workaround, just get the Crash pattern "length" down to 4 and this issue will be workarounded for you.

**Comment by SaschaH87** [comment]
Hey that sounds like a good conclusion. I willl try out.

Thanks again.

**Comment by SaschaH87** [comment]
Nice it works. 

But when i made the length first to 1 then it starts stuttering a bit, but on 4 it runs perfect.
So just one question, how can i find out in future projects which length is correct?

Thanks

**Comment by SaschaH87** [comment]
Cool, this would be great.

Thanks for your answer and for your good help too.

Bye

**Comment by mauser** [comment]
To make that clear: The silence is not considered as a bug, this is the expected behavior. 

The bug is, that the silence does **not** appear on the earlier patterns. I will have a look at this behavior in the debugger to see why this happens, i can't explain it at the moment..

You should make shure that all pattern which will play at the same time (the same column in the song editor grid) have the same length. This should help you to avoid the problem in the future.

**Comment by SaschaH87** [comment]
OK,

Now i understand and i'm pretty glad that i can go on with my work. `:)` 👍 

Again, thanks for the good support and help.


Bye

**Comment by theGreatWhiteShark** [comment]
Hey @SaschaH87 ,

Does this bug still occurs in version 1.1.1 of Hydrogen? Unfortunately the links you supplied above are dead by now and I can't have a look. Could you upload them again.

Thanks!

P.S. there is now a visual feedback when having multiple patterns with differing lengths in a single column. Hope this makes you free to use whatever pattern length combinations you want to use.

**Comment by theGreatWhiteShark** [comment]
Closed due to age. Please reopen it if the bug still occurs in current versions of Hydrogen
